lambda_src/drivers/process_https.py

```diff
@@ -59,22 +58,28 @@ def process_row(
     req_headers.setdefault('User-Agent', 'GEFF 1.0')
     req_headers.setdefault('Accept-Encoding', 'gzip')
 
+    # We look for an auth header and if found, we parse it from its encoded format
     if auth is not None:
         auth = decrypt_if_encrypted(auth)
-        assert auth is not None
+
         req_auth = (
             loads(auth)
-            if auth.startswith('{')
+            if auth and auth.startswith('{')
             else parse_header_dict(auth)
             if auth
             else {}
         )
+        auth_host = req_auth.get('host')
 
-        if 'host' in req_auth and not req_host:
-            req_host = req_auth['host']
-
-        if req_auth.get('host') != req_host:
-            pass  # if host in ct, only send creds to that host
+        # We reject the request if the 'auth' is present but doesn't match the pinned host.
+        if auth_host and auth_host != req_host:
+            raise ValueError(
+                "Requests can only be made to host provided in the auth header."
+            )
+        # We make unauthenticated request if the 'host' key is missing.
+        elif not auth_host:
+            LOG.info(f"'auth' missing the 'host' key.")
```

Comment:
these logs go to CloudWatch, which isn't going to get sent to the user. agreed with sentiment that the silent failure can be bad UX, so maybe returning the error that auth credentials must pin a host is better?